R/scale-.r

```diff
@@ -473,7 +473,13 @@ ScaleContinuous <- ggproto("ScaleContinuous", Scale,
   transform = function(self, x) {
      new_x <- self$trans$transform(x)
      if (any(is.finite(x) != is.finite(new_x))) {
-       type <- if (self$scale_name == "position_c") "continuous" else "discrete"
+       type <- if (self$scale_name == "position_c") {
+         "continuous"
+       } else if (self$scale_name == "position_d") {
+         "discrete"
+       } else {
+         "binned"
```

Comment:
It feels a little weird for "binned" to be the default. Would it be better to throw an error if the `scale_name` was an unknown value?

Reply:
I don't think we can throw an error since this only looks at position scale names... I've changed the ordering so that it defaults to discrete as before

Reply:
Why can't it be an error? It would only be seen by developers.

Reply:
The check is in ScaleContinuous, not ScalePositionContinuous... just seems weird to enforce a position specific name at this place

Reply:
I'm going to refuse this as it would require a huge rewrite unrelated to this PR. We could track this in another issue, but honestly it doesn't feel important enough for its trouble